pkg/florist/user.go

```diff
@@ -3,37 +3,66 @@ package florist
 import (
 	"fmt"
 	"log/slog"
-	"os"
 	"os/exec"
 	"os/user"
+	"strconv"
 	"strings"
 )
 
-// Add user and create home directory.
-// Do nothing if user already present.
-// Password login is disabled (use SSH public key or use passwd)
-func UserAdd(username string) error {
+// UserAddOpts contains the options to be passed to [UserAdd].
+type UserAddOpts struct {
+	// Create a system account. Default: create a user account.
+	System bool
+	// The home directory. Default: /home/<username>
+	HomeDir string
+	// A list of supplementary groups to which the user will be added.
+	// The default is for the user to belong only to the initial group.
+	Groups []string
+	// The UID. Default: choosen by the system. Use [Ptr] to fill it.
```

Comment:
what is the reason for using the pointer instead of int for a simple variable like UID?

Reply:
It is client code that will need to use Ptr:
```
	err := florist.UserAdd(name, &florist.UserAddOpts{
		UID:     florist.Ptr(42)
	})
```

Reply:
Oh... its very clear now :)

I totally missed:
> Needed to fill a struct field when the Go zero value cannot be used.

Thanks